From f2a62bc8e5fd3b022e3fd077dadca047387dddc6 Mon Sep 17 00:00:00 2001
From: Ashay Jaiswal <ashayj@codeaurora.org>
Date: Wed, 17 Jun 2015 10:42:00 +0530
Subject: [PATCH 299/959] power: qpnp-smbcharger-mmi: fix battery status
 reporting

Charger driver reports battery FULL status based on the
BAT_TCC_REACHED_RT_STS bit, there is a possibility of
this bit remaining set even after removal of charger
resulting in reporting of FULL status with no input present.

Fix this by reporting DISCHARGING if no input is present
before checking for BAT_TCC_REACHED_RT_STS bit.

CRs-Fixed: 851399
Change-Id: Ice43e758e0ecf31ac0ec3ccff0578a4558b77b31
Signed-off-by: Ashay Jaiswal <ashayj@codeaurora.org>
---
 drivers/power/qpnp-smbcharger-mmi.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index fb718aa777f..30ff034e5c8 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -841,6 +841,10 @@ static int get_prop_batt_status(struct smbchg_chip *chip)
 	bool charger_present, chg_inhibit;
 	int batt_soc;
 
+	charger_present = is_usb_present(chip) | is_dc_present(chip);
+	if (!charger_present)
+		return POWER_SUPPLY_STATUS_DISCHARGING;
+
 	batt_soc = get_prop_batt_capacity(chip);
 
 	rc = smbchg_read(chip, &reg, chip->chgr_base + RT_STS, 1);
@@ -857,10 +861,6 @@ static int get_prop_batt_status(struct smbchg_chip *chip)
 	    !chip->demo_mode && (chip->temp_state == POWER_SUPPLY_HEALTH_GOOD))
 		return POWER_SUPPLY_STATUS_FULL;
 
-	charger_present = is_usb_present(chip) | is_dc_present(chip);
-	if (!charger_present)
-		return POWER_SUPPLY_STATUS_DISCHARGING;
-
 	chg_inhibit = reg & CHG_INHIBIT_BIT;
 	if (chg_inhibit)
 		dev_err(chip->dev, "Charge Inhibit Set\n");
-- 
2.11.0

